test/integration-serial/serial_deploy_test.rb

```diff
@@ -503,6 +501,25 @@ def test_apply_failure_with_sensitive_resources_hides_template_content
     refute_logs_match("kind: Deployment") # content of the sensitive template
   end
 
+  def test_global_deploy_task
+    Krane::FormattedLogger.expects(:build).returns(@logger)
+    global_deploy = ::Krane::GlobalDeployTask.new(
+      context: task_config.context,
+      filenames: [fixture_path('globals')],
+      global_timeout: 300,
+      selector: Krane::LabelSelector.parse('app=krane')
```

Comment:
Hmm, I'd like to write a few more tests before we try and make them parallel. To do it right now I think we'd have to inject the selector, update the name to be unique and then have the ensure blocks delete by selector. Which I can't decide if its a good pattern or one that would be frustrating.

Reply:
What you're describing does sound like what we'll need to do (I don't see how it could change in the future), and we should have a test helper that takes care of it for future contributors to this task. In my opinion, this PR is responsible for introducing a sustainable way to write tests for its new task. What I'd like to see is a new `test/integration/global_deploy_test.rb` like we have for all the other tasks, with similar kinds of integration test coverage. I'm not sure whether it actually makes sense for the new suite to inherit from our regular integration test superclass, since the setup it needs is different (needs a unique partition key, but not a namespace)